cpp/examples/tpch/dbgen.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/io/parquet.hpp>
+
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/managed_memory_resource.hpp>
+#include <rmm/mr/device/owning_wrapper.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+#include <rmm/mr/device/statistics_resource_adaptor.hpp>
+
+#include <cudf_benchmark/tpch_datagen.hpp>
+
+/**
+ * @brief Log the peak memory usage of the GPU
+ */
+class memory_stats_logger {
```

Comment:
We also already have `class memory_stats_logger`.

Reply:
Yeah, its in the benchmark fixture, but since `benchmark_fixture.hpp` is not a public header, I can't import it into the `tpch/dbgen.cpp`. That's why I just copied it for now.